src/features/projectsV2/ProjectsMap/microComponents/ProjectSiteList.tsx

```diff
@@ -0,0 +1,47 @@
+import { SetState } from '../../../common/types/common';
+import styles from '../SiteDropdown.module.scss';
+
+type SiteData = {
+  siteName: string;
+  siteArea: number;
+  id: number;
+};
+interface ProjectSiteListProps {
+  siteList: SiteData[];
+  setSelectedSite: SetState<number>;
+  setIsMenuOpen: SetState<boolean>;
+  selectedSiteData: SiteData;
+}
+const ProjectSiteList = ({
+  siteList,
+  setSelectedSite,
+  setIsMenuOpen,
+  selectedSiteData,
+}: ProjectSiteListProps) => {
+  const handleSiteSelection = (index: number) => {
+    setIsMenuOpen(false);
+    setSelectedSite(index);
+  };
+  return (
+    <ul className={styles.siteListOptions}>
+      {siteList.map((site, index) => {
+        return (
+          <li
+            className={`${styles.listItem} ${
+              site.siteName === selectedSiteData?.siteName
+                ? styles.selectedItem
+                : ''
+            }`}
```

Comment:
When there are multiple sites with the same name, this logic does not work. Always use site id to decide which site is selected as we know that is unique.

For example, for the following link: [/prd/forest-restoration-program?site=site_NekKEGqkIO4rZ5C](https://planet-webapp-multi-tenancy-setup-git-projects-9fa0c1-planetapp.vercel.app/en/prd/forest-restoration-program?site=site_NekKEGqkIO4rZ5C)